backend/src/hatchling/cli/build/__init__.py

```diff
@@ -72,7 +75,21 @@ def build_impl(
         if not (clean_only or show_dynamic_deps) and len(target_data) > 1:
             app.display_mini_header(target_name)
 
-        builder = builder_class(root, plugin_manager=plugin_manager, metadata=metadata, app=app.get_safe_application())
+        variant_build_kwargs = {}
+        if f'{builder_class.__module__}.{builder_class.__name__}' == 'hatchling.builders.wheel.WheelBuilder':
```

Comment:
I honestly doubt such a hack will be accepted upstream.

Reply:
How should we proceed? I can think of a few options here.

1. Import WheelBuilder and use `isinstance` to check the type
2. Pass variant kwargs unconditionally to all builders. This will require updating all builder classes to handle the kwargs.

Reply:
I think the latter will be a better fit.